db/schema_tables.cc

```diff
@@ -1774,7 +1780,11 @@ static shared_ptr<cql3::functions::user_aggregate> create_aggregate(replica::dat
 
     bytes_opt initcond = std::nullopt;
     if (initcond_str) {
-        initcond = state_type->from_string(initcond_str.value());
+        auto expr = cql3::util::do_with_parser(*initcond_str, std::mem_fn(&cql3_parser::CqlParser::term));
```

Comment:
Should you catch parse error exceptions and convert them into a more useful exception, or is the parse error exception readable enough? Can you please add a test for this case (an unparsable value for an initializer)?

Reply:
It would be hard to create such a test, because the string should come from a value that has already been parsed once. The issue could arise if the `to_parsable_string()` returned an incorrect string, and I'm not sure if that method has been extensively tested (it's one of the main methods used by the `expression::printer`)

Reply:
Good point. I was under the mistaken impression that the initializer is a string that the user types as a CQL string, but I see now it's actually a CQL expression, so if it was malformed we would have had  a parse error in the original statement, and would have never saved a malformed expression.